src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -187,7 +187,8 @@ class CallActor(key: CallKey, locallyQualifiedInputs: CallInputs, backend: Backe
     )
 
     val message = executionResult match {
-      case SuccessfulExecution(outputs, returnCode, hash) => WorkflowActor.CallCompleted(key, outputs, returnCode, hash)
+      case SuccessfulExecution(outputs, returnCode, hash) =>
+        WorkflowActor.CallCompleted(key, outputs, returnCode, if (workflowDescriptor.writeToCache) hash else "")
```

Comment:
It's not a complicated topic, you already have all the info. One either agrees with it or not and you' don't, which is perfectly fine.

You're completely correct that if you incorrectly wrap a value you've gained nothing. However what you *have* done is move the places where accidental code mistakes can happen only to when those values enter in to the system instead of every single function in the entire codebase where that primitive is an argument. Iff the value is correctly wrapped at the creation/ingestion point you have compile time guarantees that they're not being passed in to the wrong places throughout the system. Note that if you didn't wrap it you'd have the exact same issue but also have no guarantees that you didn't accidentally change some code e.g. swapping the order of string arguments to a function.

Also note that if you make these value classes that there's no overhead at the byte code level, the case classes are purely in code and don't exist post compilation.